enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,248 @@
+# openshift scc runtime class enhancement
+---
+title: gate-runtime-classes-with-scc
+authors:
+  - "@haircommander"
+  - "@mrunalp"
+reviewers:
+  - "@deads2k"
+  - "@sttts"
+approvers:
+  - ""
+creation-date: 2020/12/04
+last-updated: 2020/12/11
+status: provisional
+
+---
+# Gate Runtime Classes with SCC
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enhance the Openshift SCC with two fields called `AllowRuntimeClasses` and `EnsureRuntimeClasses`, which gate users from specifying runtime classes in their Pod requests. 
+
+## Motivation
+
+Enhancing SCC to be runtime class aware allows admins to customize how their users are able to access different runtime customizations. This is key in enabling new features like unprivileged builds, kata containers, and performance enhancements for latency sensitive workloads, while also providing protection from less privileged users gaining access to features they should not.
+
+### Goals
+
+- SCC should support specifying the runtime classes that a user is allowed access to in their pod specs.
+- SCC should support specifying the runtime classes that must be specified in the pod request from a user.
+
+### Non-Goals
+
+- Changing the current SCC profiles to allow any extra runtime classes.
+- Changing behavior of the default runtime class.
+- Adding additional SCCs for various customzied behavior using AllowedRuntimeClasses.
+
+## Proposal
+
+
+[RuntimeClasses](https://kubernetes.io/docs/concepts/containers/runtime-class/) is a feature in Kubernetes that allows a user to request a certain runtime configuration. CRI-O has enhanced this feature to not only branch on different runtimes, but customize the way the pod is run. With the customization of the way a pod is run comes risk of users allocating resources they should not be allowed to. This leads to the need for a way to gate different users from using different runtime classes, and SCC is a perfect component to do so.
+
+The proposal is to add two different fields to the SCC: `AllowRuntimeClasses` and `EnsureRuntimeClasses`.
+
+### AllowRuntimeClasses
+The API for `AllowRuntimeClasses` will follow the implementation of [AllowCapabilities](https://github.com/openshift/api/blob/0ba2c3658da6cade27a5066575cd8446eb03914c/security/v1/types.go#L57..L61), where a specific SCC profile will be able to specify runtime classes that are allowed for a Pod. If a request for a runtime class comes in that doesn't match the allowed runtime classes, the Pod will be denied.
```

Comment:
and empty string would be specified in the list to indicate, "allow pods without a runtimeclass set"?